app/api/issues/edit/[id]/route.ts

```diff
@@ -0,0 +1,25 @@
+import { editIssueSchema } from '@/features/issue/schemas'
+import { NextRequest, NextResponse } from 'next/server'
+import prisma from '@/prisma/db'
+import { z } from 'zod'
+import { Status } from '@prisma/client'
+
+export const PATCH = async (request: NextRequest) => {
+	const id = parseInt(request.url.split('/').slice(-1)[0])
+	const body = (await request.json()) as {
+		issue: z.infer<typeof editIssueSchema>
+	}
+	const editedIssue = body.issue
+
+	if (isNaN(id)) {
+		return NextResponse.json(
+			{},
+			{ status: 400, statusText: `Id: '${id}' is not a number` }
+		)
+	}
+	const issue = await prisma.issue.update({
+		where: { id },
+		data: { ...editedIssue, status: Status[editedIssue.status] },
```

Comment:
Widze ze enuma uzywasz w editIssueSchema jako z.enum([]). Może być ale lepsze jest z.nativeEnum(Status), bardziej czytelne i wygodniejsze

Reply:
I nie używaj enuma z prismy, tylko tego co masz w pliku IssueModel

Reply:
I wyjeb to z typu IssueModel [x: string]: any, po co to?

Reply:
I ten plik IssueModel do zastanowienia się czy nie powinien być w /features/issue/types zamiast w globalnym types

Reply:
`export const EditIssueSchema = z.object({
	title: z.string().min(1).max(255),
	description: z.string().min(1),
	status: z.nativeEnum({
		OPEN: 'OPEN',
		IN_PROGRESS: 'IN_PROGRESS',
		CLOSED: 'CLOSED',
	}),
})` 
To jest ta schema z native enum

`const editedIssue = await prisma.issue.update({
			where: { id },
			data: { ...issue, status: Status[issue.status] },
		})`

I nie działa mi w tym miejscu  Status[issue.status] lement implicitly has an 'any' type because expression of type 'string' can't be used to index type 'typeof Status'.
  No index signature with a parameter of type 'string' was found on type 'typeof Status'.ts(7053)